### PR TITLE
[ticket/14431] Remote avatar uploading (master changes)

### PR DESCRIPTION
**Pull request by exussum12**
replace the 3.1 way with guzzle
Adds guzzle as a dependency
Removed a test as it wont wont with guzzle (as far as I know)

### CODE REVIEW

**Comment by exussum12** [comment]
These checks fail on master too. I'll submit another pull request for the 3.1 branch

**Comment by exussum12** [comment]
Updated to use guzzle adds guzzle as a dependency

**Comment by exussum12** [comment]
The failing test now should probably be changed too, current test is assuming no redirect. Is there anywhere we can host images for the tests ?

File: phpBB/phpbb/files/types/remote.php [hunk 1]
```diff
@@ -101,103 +97,32 @@ protected function remote_upload($upload_url)
...
+			return $this->factory->get('filespec')->set_error($this->language->lang($this->upload->error_prefix . 'NOT_UPLOADED'));
+		} catch (\Guzzle\Http\Exception\RequestException $requestException) {
+			return $this->factory->get('filespec')->set_error($this->upload->error_prefix . 'REMOTE_UPLOAD_TIMEOUT');
+		} catch (\Exception $e) {
 			return $this->factory->get('filespec')->set_error($this->language->lang($this->upload->error_prefix . 'NOT_UPLOADED'));
 		}
 
-		// Make sure $path not beginning with /
-		if (strpos($path, '/') === 0)
+		if ($remote_max_filesize && $response->getContentType() > $remote_max_filesize)
```
Comment (marc1706) [inline phpBB/phpbb/files/types/remote.php]: Needs to be ```$response->getContentLength()```

File: phpBB/phpbb/files/types/remote.php [hunk 2]
```diff
@@ -101,103 +97,32 @@ protected function remote_upload($upload_url)
...
-			// Cancel upload if we exceed timeout
-			if (!empty($stream_meta_data['timed_out']) || time() >= $timer_stop)
-			{
-				return $this->factory->get('filespec')->set_error($this->upload->error_prefix . 'REMOTE_UPLOAD_TIMEOUT');
-			}
+			return $this->factory->get('filespec')->set_error($this->language->lang($this->upload->error_prefix . 'WRONG_FILESIZE', $max_filesize['value'], $max_filesize['unit']));
 		}
-		@fclose($fsock);
 
-		if (empty($data))
```
Comment (marc1706) [inline phpBB/phpbb/files/types/remote.php]: You should still check if the content length is 0.

**Comment by marc1706** [comment]
You need to update the composer lock file by running ```php ../composer.phar update nothing``` in the phpBB folder.

**Comment by exussum12** [comment]
Didnt run the composer update due to conflicts. Added it here by request.

Fixed the other errors too thanks

**Comment by marc1706** [comment]
I'll take this over and fix the composer conflicts.

**Pull request closed without merging**